src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -805,7 +811,10 @@ public static class Builder {
 
         // Optional fields
         private String sourceTag;
-        private OkHttpClient okHttpClient = new OkHttpClient();
+        private ProxyConfig proxyConfig;
+        private final OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder();
+        private OkHttpClient customOkHttpClient = new OkHttpClient();
```

Comment:
Why do we need to differentiate building an Ok client internally vs a custom one? 

At the very least it feels a little awkward to me to pre-instantiate both options.

Reply:
If somebody did this they'd get back unexpected behavior compared to the alternate order. Seems like it should be an error -- you shouldn't be able to set an Ok client twice.

```java
Pinecone pc = new Pinecone.Builder()
    .withProxy(...)
    .withOkHttpClient(...)
    .build();
```

Reply:
1. We should not instantiate the Builder and the customOkHttpClient.
2. You are right, if the order of operations is .withProxy(...).withOkHttpClient(...), then the proxy config will be ignored and the custom OkHttpClient will be set. My intention was to prevent user from setting both the custom client and configuring proxy so what I can do is throw an error if both are called. Thoughts?

Reply:
yeah, that would be preferable I think vs. having them get unexpected behavior at runtime